### ISSUE TITLE
Does not compile with upstream libdragon by default

### ISSUE THREAD

**Issue by gameblabla** [networkfusion/alt64#31]
As the title suggests, currently it does not compile as is with upstream libdragon.
I have a few branches and workarounds to make it work with upstream, see here :
https://github.com/gameblabla/altra64/tree/upstream_libd_fix
https://github.com/gameblabla/altra64/tree/fixmakefile_upstream
https://github.com/gameblabla/altra64/commit/29a6b71415318369141462e1cf363f6638fb0688

However, i noticed in one of your pull requests that you intend on using an old fork of conker's libdragon. Do you intend on sticking with it ?

**Comment by networkfusion** [comment]
Yes the version in n64-tools, is is a hybrid of the official libdragon, conkers version and my own. I dont like the way the official version has to build GCC in order to build the lib and their new docker build is irrelevant... I have split them and use automated CICD for each (which is how it should be done). I may have a switch that supports both in future, or even switch to n64chain but that will only come once I start on the UI again... If you see any bug fixes in the official version that would benefit in my custom one, feel free to point them out.

**Comment by gameblabla** [comment]
I should mention that libdragon's upstream code still has the toolchain script :
https://github.com/DragonMinded/libdragon/blob/master/tools/build
In fact that's how i built it because docker was not working for me.

Conker's libdragon doesn't have stuff like the ones below
https://github.com/DragonMinded/libdragon/pull/41
https://github.com/DragonMinded/libdragon/pull/37
https://github.com/DragonMinded/libdragon/pull/60

It is very behind upstream which has seen several fixes since then. It would be nice to merge libconker's libdragon changes if you need any changes from that... But didn't he merge those anyway ?

**Comment by networkfusion** [comment]
Ok, I will think about those changes. If some of conkers changes were rolled into the main libdragon repo I would probably swap back... but I may also make the version optional... but since the build process is automated and the download is available for both platforms, I don't see it as a showstopper and a very nice optimisation...

**Comment by networkfusion** [comment]
I have now added most of the relevant libdragon changes (c++ single entry point still causes issues, so I am still wondering how to handle it). I will also consider how the changes from conker64 can be made into a pull request to the main repo, so that way I can change back... the main differences are around RSP being implemented very differently... but altra64 doesnt use it, so may just be a non issue...
